### PR TITLE
bf(ZENKO-1782): Fix flaky grafana tests

### PR DESCRIPTION
**Pull request by tmacro**
Add retry to work around network failures

### CODE REVIEW

**Review by ssalaues — APPROVED** [review]

File: tests/zenko_tests/python_tests/zenko_e2e/grafana/client.py [hunk 1]
```diff
@@ -8,20 +10,36 @@ def __init__(self, user, password, endpoint):
...
+        for _ in range(4):  # One try with 3 retries
+            try:
+                resp = requests.get(
+                    path,
+                    auth=HTTPBasicAuth(self._user, self._password)
+                )
+                assert resp
+                assert resp.status_code == 200
+                assert resp.json()
+                return True
```
Comment (jonathan-gramain) [inline tests/zenko_tests/python_tests/zenko_e2e/grafana/client.py:24]: Don't you want to `return resp.json()` not to change the previous behavior?

File: tests/zenko_tests/python_tests/zenko_e2e/grafana/client.py [hunk 2]
```diff
@@ -8,20 +10,36 @@ def __init__(self, user, password, endpoint):
...
+            except Exception:  # pylint: disable=broad-except
+                pass
+            time.sleep(backoff)
+            backoff *= 2
+        return False
```
Comment (jonathan-gramain) [inline tests/zenko_tests/python_tests/zenko_e2e/grafana/client.py:29]: you might `assert.fail` here (same reason not to change the previous behavior)

File: tests/zenko_tests/python_tests/zenko_e2e/grafana/client.py (hunk 1 above)
Comment (tmacro) [reply tests/zenko_tests/python_tests/zenko_e2e/grafana/client.py:24]: It's only the grafana tests that use this so I think a  little rework is OK, also it keeps our requests checking in one place rather that duplicating it

**Comment by tmacro** [comment]
@bert-e reset

**Comment by tmacro** [comment]
@bert-e approve

**Review by jonathan-gramain — APPROVED** [review]

**Comment by tmacro** [comment]
@bert-e reset

**Comment by tmacro** [comment]
@bert-e reset

**Pull request merged**